### PR TITLE
Fixes for type inference and improvements to gradio.

### PR DESCRIPTION
**Pull request by lcrh**

### CODE REVIEW

File: src/enact/gradio/gradio.py [hunk 1]
```diff
@@ -340,12 +340,12 @@ def __init__(
         input required exception.
```
Comment (isaacery) [inline src/enact/gradio/gradio.py:340]: typo: input_rquired_outputs

File: src/enact/gradio/gradio.py [hunk 2]
```diff
@@ -340,12 +340,12 @@ def __init__(
...
     """
```
Comment (isaacery) [inline src/enact/gradio/gradio.py:341]: nit: Above `Type[Any]` could just be `Type`

File: src/enact/gradio/gradio.py [hunk 3]
```diff
@@ -340,12 +340,12 @@ def __init__(
...
     self._invokable = invokable
-    self._input_required_inputs = input_required_inputs or [
-      str, type(None), int, float, bool, PIL.Image.Image
-    ]
-    self._input_required_outputs = input_required_outputs or [
-      str, type(None), int, float, bool, PIL.Image.Image
-    ]
+    self._input_required_inputs = cast(List[Type], list({
+      str, type(None), int, float, bool, PIL.Image.Image}.union(
```
Comment (isaacery) [inline src/enact/gradio/gradio.py:344]: Factor this set out into a constant.

**Review by isaacery — APPROVED** [review]

File: src/enact/gradio/gradio.py (hunk 1 above)
Comment (lcrh) [reply src/enact/gradio/gradio.py:344]: Done

File: src/enact/gradio/gradio.py (hunk 2 above)
Comment (lcrh) [reply src/enact/gradio/gradio.py:345]: Done

File: src/enact/gradio/gradio.py (hunk 3 above)
Comment (lcrh) [reply src/enact/gradio/gradio.py]: Done

**Pull request merged**